packages/tenderly-hardhat/src/utils/util.ts

```diff
@@ -1,13 +1,220 @@
+    }
+    logger.trace(`ChainId for network '${network}' is ${chainId}`);
+
+    if (chainId === undefined) {
+      logger.error(
+        `Error in ${PLUGIN_NAME}: Couldn't identify network. Please provide a chainId in the network config object`
+      );
+      return null;
+    }
+
+    contracts.push({
+      contractToVerify: flatContract.name,
+      // TODO(dusan) this can be done with TASK_COMPILE_SOLIDITY_GET_COMPILER_INPUT hardhat task
+      sources: await extractSources(job),
+      compiler: job.getSolcConfig(),
+      networks: {
+        [chainId]: {
+          address: flatContract.address,
+          links: flatContract.libraries,
+        },
+      },
+    });
+  }
+  // TODO(dusan) see about merging compilation jobs?
+
+  return {
+    contracts,
+  };
+}
+
+async function extractSources(job: CompilationJob): Promise<Record<string, TenderlyVerifyContractsSource>> {
+  const sources: Record<string, TenderlyVerifyContractsSource> = {};
+  logger.info("Extracting sources from compilation job.");
+
+  for (const file of job.getResolvedFiles()) {
+    const name = file.sourceName.split("/").slice(-1)[0].split(".")[0];
+    sources[file.sourceName] = {
+      name,
+      code: file.content.rawContent,
+    };
+  }
+
+  return sources;
+}
+
+// TODO(dusan): This function shouldn't be here. Fork verification should have the same request format as private or public verification
+async function _convertToForkRequest(
+  hre: HardhatRuntimeEnvironment,
+  request: TenderlyVerifyContractsRequest
+): Promise<TenderlyForkContractUploadRequest> {
+  const forkRequest: TenderlyForkContractUploadRequest = { config: {}, contracts: [], root: "" };
+
+  // Merge all compilation jobs to see if they can all fit into one request, if not, throw an error because fork verification doesn't support multiple compiler version.
+  const compilationJobs: CompilationJob[] = [];
+  for (const contract of request.contracts) {
+    compilationJobs.push(await getCompilationJob(hre, contract.contractToVerify));
+  }
+  const mergedJob: CompilationJob[] = await hre.run(TASK_COMPILE_SOLIDITY_MERGE_COMPILATION_JOBS, { compilationJobs });
+  if (mergedJob.length > 1) {
+    throw new Error(
+      "The provided contracts must be compiled with the same compiler configuration. At this point, hardhat-tenderly plugin doesn't support multi compiler verification on FORKS. Private and public verification support multi compiler verification."
+    );
+  }
+
+  for (const contract of request.contracts) {
+    for (const [sourcePath, source] of Object.entries(contract.sources)) {
+      const forkContract: TenderlyContract = {
+        compiler: undefined,
+        contractName: source.name,
+        networks: undefined,
+        source: source.code,
+        sourcePath,
+      };
+      if (
+        forkContract.contractName === contract.contractToVerify ||
+        `${forkContract.sourcePath}:${forkContract.contractName}` === contract.contractToVerify
```

Comment:
Can we make a function that is fetching this string value?

(e.g. `spawnSourcePathContractNameKey()`)

Reply:
Isn't this more readable? Functions change code inspection flow, the reader would have to go to a definition of a one-line simple function and go back instead of just passing through the code.
I get the reason behind `extractFileName`, but if the reason for this change is readability, isn't this more readable?

Reply:
It would be less error-prone (if there is a change in the future for sourcePath generation) if we consider maintaining parity in the generation and spawning of the sourcePath across the hardhat-plugin. By keeping this code block in a single location, we can ensure consistency and avoid duplication. However, if this code block is already in a single place, then it is fine for now.